kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue

```diff
@@ -125,29 +101,78 @@
     components: {
       'icon-button': require('kolibri.coreVue.components.iconButton'),
     },
-    props: [
-      'userid', 'username', 'fullname', 'roles', // TODO - validation
-    ],
+    props: {
+      userid: {
+        type: String, // string is type returned from server
+        required: true,
+      },
+      fullname: {
+        type: String,
+        required: true,
+      },
+      username: {
+        type: String,
+        required: true,
+      },
+      userkind: {
+        type: String,
+        required: true,
+      },
+    },
     data() {
       return {
         username_new: this.username,
         password_new: '',
         password_new_confirm: '',
         fullName_new: this.fullname,
-        role_new: this.roles.length ? this.roles[0].kind : 'learner',
+        kind_new: this.userkind,
```

Comment:
ah, I didn't notice `kind_new` can be assigned new values here `v-model="kind_new"`. you are right, we should treat prop as immutable.